src/support/compare.ts

```diff
@@ -459,6 +447,41 @@ function diffPlainObject(a: any, b: any, options: DiffOptions): (ConstructRecord
 	return patchRecords;
 }
 
+export function getComparableObjects(a: any, b: any, options: DiffOptions) {
+	const { ignoreProperties = [], ignorePropertyValues = [] } = options;
+	const ignore = new Set<string>();
+	const keep = new Set<string>();
+
+	function isIgnoredProperty(name: string) {
```

Comment:
Small nit, but couldn't we create a function that wouldn't have to check `Array.isArray()` on each invocation, since it is not going to change types during the whole of the out function, right?